lib/src/rules/avoid_function_literal_in_foreach_method.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.avoid_function_literal_in_foreach_method;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Avoid using Iterable.forEach() with a function literal.';
+
+const _details = r'''
+
+**AVOID** using Iterable.forEach() with a function literal.
+
+**BAD:**
+```
+people.forEach((person) {
+  ...
+});
+```
+
+**GOOD:**
+```
+for (var person in people) {
+  ...
+}
+```
+
+**GOOD:**
+```
+for (var person in people) {
+  ...
+}
+```
+people.forEach(print);
+''';
+
+class AvoidFunctionLiteralInForeachMethod extends LintRule {
+  _Visitor _visitor;
+  AvoidFunctionLiteralInForeachMethod() : super(
+          name: 'avoid_function_literal_in_foreach_method',
+            description: _desc,
+            details: _details,
+            group: Group.style){
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  visitMethodInvocation(MethodInvocation node) {
+    if (node.target != null &&
+      DartTypeUtilities.implementsInterface(node.target.bestType, 'Iterable', 'dart.core') &&
```

Comment:
The type check is the most expensive part of the condition, and hence should be moved to the end.